web/Web/web/search/templates/search/search_list.html

```diff
@@ -76,3 +76,29 @@ <h6 class="title" style="margin-bottom: 0;">{{ hit.title|safe }}</h6>
   </div>
   {% endif %}
 {% endif %}
+
+<nav aria-label="...">
+  <ul class="pagination justify-content-center">
+      <li class="page-item {{ pagination.is_first_page|yesno:'disabled,' }}">
+      <a class="page-link" href="{% url 'search:query_result' query_id=queryID%}?page_number=1">First</a>
```

Comment:
If you go to the search page without any queryID (e.g. http://localhost/search/), django will complain 

```
NoReverseMatch at /search/
Reverse for 'query_result' with keyword arguments '{'query_id': 'NA'}' not found. 1 pattern(s) tried: ['search\\/query\\/(?P<query_id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\\/$']
```

Reply:
My bad. should have done a better job testing.